packages/aragon-wrapper/src/index.js

```diff
@@ -923,12 +927,17 @@ export default class Aragon {
    * @return {Promise} Resolves with the identity or null if not found
    */
   resolveAddressIdentity (address) {
-    const providerName = 'local' // TODO - get provider
-    const provider = this.identityProviderRegistrar.get(providerName)
-    if (provider && typeof provider.resolve === 'function') {
-      return provider.resolve(address)
-    }
-    return Promise.reject(new Error(`Provider (${providerName}) not installed`))
+    const providerNames = [ 'local', 'addressBook' ] // TODO - get provider
+    return providerNames.reduce(async (identity, providerName, index) => {
+      if (await identity) {
+        return identity
```

Comment:
This feels backward to me. We know `identity` will be a promise that resolves with `null` on the first pass. So that means that, if `LocalIdentityProvider` does resolve to an identity, we don't notice that until we're into the second pass of the loop, which is mostly concerned with checking on `AddressBookIdentityProvider`.

Maybe we could rewrite this something like:

```js
async resolveAddressIdentity (address) {
  const providerNames = [ 'local', 'addressBook' ] // TODO - get provider
  const resolvedIdentities = await Promise.all(
    providerNames.map(providerName => {
      const provider = this.identityProviderRegistrar.get(providerName)
      if (provider && typeof provider.resolve === 'function') {
        return provider.resolve(address)
      }
      return Promise.reject(new Error(`Provider (${providerName}) not installed`))
    })
  )
  return resolvedIdentities.find(x => !!x) // return first non-null
}
```

Note that using `Promise.all` like this might hold things up waiting for identity resolutions from irrelevant sources – that is, if there is an identity in custom labels, we're going to spend cycles looking up the identity in AddressBook for no reason. If we don't like that, I think the cleanest approach might be to use [a `for..of` loop](https://stackoverflow.com/a/32101207/249801).

Reply:
Also note that the `find(x => !!x)` could be rewritten as `find(x => x)` or `find(x => Boolean(x))` depending on the style guide of the repo.